esx_service/vmdk_ops.py

```diff
@@ -640,6 +643,22 @@ def dev_info(unit_number, pci_slot_number):
     return {'Unit': str(unit_number),
             'ControllerPciSlotNumber': pci_slot_number}
 
+def reset_vol_meta(vmdk_path):
+    '''Clears metadata for vmdk_path'''
+    vol_meta = kv.getAll(vmdk_path)
+    if not vol_meta:
+       vol_meta = {}
+    logging.debug("Reseting meta-data for disk=%s", vmdk_path)
+    if set(vol_meta.keys()) & {kv.STATUS, kv.ATTACHED_VM_UUID, kv.ATTACHED_VM_NAME}:
+          logging.debug("Old meta-data for %s was (status=%s VM name=%s uuid=%s)",
+                        vmdk_path, vol_meta[kv.STATUS],
+                        vol_meta[kv.ATTACHED_VM_NAME],
+                        vol_meta[kv.ATTACHED_VM_UUID])
+    vol_meta[kv.STATUS] = kv.DETACHED
+    vol_meta[kv.ATTACHED_VM_UUID] = None
+    vol_meta[kv.ATTACHED_VM_NAME] = None
+    if not kv.setAll(vmdk_path, vol_meta):
+       logging.warning("clear_volume_status: Failed to save Disk metadata for %s", vmdk_path)
```

Comment:
We should probably return an error and refuse to do the mount on the new VM, because otherwise we will have invalid metadata showing attachment to 2 VMs.